Source/WebCore/loader/cache/CachedResourceLoader.cpp

```diff
@@ -1195,7 +1195,9 @@ ResourceErrorOr<CachedResourceHandle<CachedResource>> CachedResourceLoader::requ
                 return makeUnexpected(ResourceError { errorDomainWebKitInternal, 0, url, "Resource blocked by content blocker"_s, ResourceError::Type::AccessControl });
             }
             if (type == CachedResource::Type::MainResource && RegistrableDomain { resourceRequest.url() } != originalDomain) {
-                frame->loader().load(FrameLoadRequest { frame, { resourceRequest.url() } });
+                RunLoop::protectedMain()->dispatch([frame, url = resourceRequest.url()] {
```

Comment:
A comment for why this is needed would be good here so people know why we are delaying.